.github/workflows/images.yaml

```
@@ -57,10 +60,10 @@ jobs:
           ref: ${{ steps.tag.outputs.tag }}
 
       # main branch or tag pushes
-      - name: CI Build ${{ matrix.name }}
+      - name: Build ${{ matrix.name }}
         if: ${{ github.event_name != 'pull_request_target' }}
```

let's run step only on tag push, not branch push. https://quay.io/repository/cilium/cilium-cli?tab=tags should only contain released images 🙏 
```suggestion
        if: ${{ github.ref_type == 'tag' }}
```

---

@michi-covalent maybe `github.ref_type == 'tag' && github.event_name == 'tag'`?

---

This was never changed. Happy to change it, but didn't want to modify the current pipeline behavior.

Let me know and I can perform the update.

---

I'm not even sure if you want to change it, because this would allow the release of the image on merges to main using the commit hash.

See https://github.com/cilium/cilium-cli/pull/2782/files#diff-d0a3d6684c78a148cbf0725d5fe8b5aab6431da05b698a82c9e015516f3020baR48-R57

So changing this would change the existing behavior

---

the latest workflow file looks good to me. just wanted to make sure we don't push to quay.io/cilium/cilium-cli on push to main branch 🚀🙏

---

hmm don't we still end up pushing to quay.io/cilium/cilium-cli repo on push to main? https://github.com/cilium/cilium-cli/blob/f996cce593f6661b90ee049fdbe8e2ed6cb28224/.github/workflows/images.yaml#L64-L77 💭 

i suggest keeping this pull request for changing Dockerfile only. let's forcus on getting Dockerfile change merged in this PR. @tklauser and i will set up a release image workflow. we need to create a separate github environment anyways.

---

we'll do something like this for the release image workflow once the Dockerfile change gets merged => https://github.com/cilium/cilium-cli/pull/2882/commits/c6cfc8bebd44ddc4f7b757a2be398283c0ff1656

---

@michi-covalent the separete github env is already created based on earlier feedback.

https://github.com/cilium/cilium-cli/pull/2782/files#diff-d0a3d6684c78a148cbf0725d5fe8b5aab6431da05b698a82c9e015516f3020baR23

---

i didn't mean to update the workflow file to refer to a release environment. i need to configure a github environment for this repo and make sure that it works.

let's limit the scope of this pull request to the Dockerfile change 🙏

---

Without the workflow it won't work right. Hence https://github.com/cilium/cilium-cli/pull/2782#discussion_r1888426351

The Github environment you have to configure basically needs to match the name here https://github.com/cilium/cilium-cli/pull/2782/files#diff-d0a3d6684c78a148cbf0725d5fe8b5aab6431da05b698a82c9e015516f3020baR31

Guess the only thing you need to do is configure the secrets for publishing the image int there.

Why don't you add the environment, (give it a different name to your desires) and then we update it here in the workflow.

I'm not fully getting the problem with the workflow here. Basically all required to use a different gh-env is in place. You only need to add the secret to it.